packages/adapters-library/src/core/unwrapCache.ts

```diff
@@ -0,0 +1,89 @@
+import type {
+  IPricesAdapter,
+  PricesInput,
+} from '../adapters/prices-v2/products/usd/pricesV2UsdAdapter'
+import type { IProtocolAdapter } from '../types/IProtocolAdapter'
+import type { UnwrapExchangeRate, UnwrapInput } from '../types/adapter'
+import type { Chain } from './constants/chains'
+
+const TEN_MINUTES_IN_MS = 10 * 60 * 1000
+
+export interface IUnwrapCache {
+  fetchUnwrapWithCache(
+    adapter: IProtocolAdapter,
+    input: UnwrapInput,
+  ): Promise<UnwrapExchangeRate>
+  fetchPriceWithCache(
+    adapter: IPricesAdapter,
+    input: PricesInput,
+  ): Promise<UnwrapExchangeRate>
+}
+
+export class MemoryUnwrapCache implements IUnwrapCache {
+  private readonly cache: Map<
+    string,
+    Promise<{ value: UnwrapExchangeRate; expiryTime: number }>
+  >
+  private readonly expiryTimeInMs: number
+
+  constructor(expiryTimeInMs = TEN_MINUTES_IN_MS) {
+    this.cache = new Map()
+    this.expiryTimeInMs = expiryTimeInMs
+  }
+
+  async fetchUnwrapWithCache(
+    adapter: IProtocolAdapter,
+    input: UnwrapInput,
+  ): Promise<UnwrapExchangeRate> {
+    return await this.fetchWithCache(
+      adapter.chainId,
+      input.protocolTokenAddress,
+      !!input.blockNumber,
```

Comment:
Passing a block number ignores the cache. We do not need to maintain block number caching as we no longer pass block numbers via API.